File: src/main/java/com/interviewplannerapp/service/FocusAreaService.java
package com.interviewplannerapp.service;

import java.util.List;

import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.data.jpa.domain.Specification;
import org.springframework.http.ResponseEntity;

import com.interviewplannerapp.domain.FocusArea;
import com.interviewplannerapp.dto.FocusAreaDTO;
import com.interviewplannerapp.dto.FocusAreaSearchDTO;
import com.interviewplannerapp.dto.FocusAreaPageDTO;
import com.interviewplannerapp.dto.FocusAreaConvertCriteriaDTO;
import com.interviewplannerapp.service.GenericService;
import com.interviewplannerapp.dto.common.RequestDTO;
import com.interviewplannerapp.dto.common.ResultDTO;
import java.util.List;
import java.util.Optional;





public interface FocusAreaService extends GenericService<FocusArea, Integer> {

	List<FocusArea> findAll();

	ResultDTO addFocusArea(FocusAreaDTO focusAreaDTO, RequestDTO requestDTO);

	ResultDTO updateFocusArea(FocusAreaDTO focusAreaDTO, RequestDTO requestDTO);

    Page<FocusArea> getAllFocusAreas(Pageable pageable);

    Page<FocusArea> getAllFocusAreas(Specification<FocusArea> spec, Pageable pageable);

	ResponseEntity<FocusAreaPageDTO> getFocusAreas(FocusAreaSearchDTO focusAreaSearchDTO);
	
	List<FocusAreaDTO> convertFocusAreasToFocusAreaDTOs(List<FocusArea> focusAreas, FocusAreaConvertCriteriaDTO convertCriteria);

	FocusAreaDTO getFocusAreaDTOById(Integer focusAreaId);







}

File: interviewplannerapp_ui/src/services/FocusAreaService.js
import http from "../http-common"; 

class FocusAreaService {
  getAllFocusAreas(searchDTO) {
    console.log(searchDTO)
    return this.getRequest(`/focusArea/focusAreas`, searchDTO);
  }

  get(focusAreaId) {
    return this.getRequest(`/focusArea/${focusAreaId}`, null);
  }

  findByField(matchData) {
    return this.getRequest(`/focusArea?field=${matchData}`, null);
  }

  addFocusArea(data) {
    return http.post("/focusArea/addFocusArea", data);
  }

  update(data) {
  	return http.post("/focusArea/updateFocusArea", data);
  }
  
  uploadImage(data,focusAreaId) {
  	return http.postForm("/focusArea/uploadImage/"+focusAreaId, data);
  }




	postRequest(url, data) {
		return http.post(url, data);
      };

	getRequest(url, params) {
        return http.get(url, {
        	params: params
        });
    };

}

export default new FocusAreaService();
